modules/consent/lib/Auth/Process/Consent.php

```diff
@@ -144,13 +144,34 @@ public function __construct($config, $reserved)
     /**
      * Helper function to check whether consent is disabled.
      *
-     * @param mixed $option  The consent.disable option. Either an array or a boolean.
+     * @param mixed $option  The consent.disable option. Either an array of array, an array or a boolean.
      * @param string $entityIdD  The entityID of the SP/IdP.
      * @return boolean  TRUE if disabled, FALSE if not.
      */
     private static function checkDisable($option, $entityId) {
         if (is_array($option)) {
-            return in_array($entityId, $option, TRUE);
+            // Check if consent.disable array has one element that is an array
+            if (count($option) == count($option, COUNT_RECURSIVE)) {
+                // Array is not multidimensional.  Simple in_array search suffices
+                return in_array($entityId, $option, TRUE);
```

Comment:
Our [contribution guidelines](https://github.com/simplesamlphp/simplesamlphp/blob/master/CONTRIBUTE.md) specify the use of PSR-2 coding standard. This means boolean constants, `null` and others must be lowercase. So s/TRUE/true/g.